airflow/jobs/scheduler_job_runner.py

```diff
@@ -1847,3 +1851,23 @@ def _executor_to_tis(self, tis: list[TaskInstance]) -> dict[BaseExecutor, list[T
             _executor_to_tis[ExecutorLoader.load_executor(executor)].append(ti)
 
         return _executor_to_tis
+
+    @provide_session
+    def _check_and_activate_dagrun_scheduling(self, session: Session = NEW_SESSION):
+        tis = session.scalars(select(TI).where(TI.state == TaskInstanceState.UP_FOR_RETRY)).all()
+        self.log.debug("Checking for DagRuns ready for scheduling")
+        drs = []
+        for ti in tis:
+            if ti.dag_run in drs:
+                continue
+            # Get task from the Serialized DAG
+            try:
+                dag = self.dagbag.get_dag(ti.dag_id)
+                ti.task = dag.get_task(ti.task_id)
+            except Exception:
```

Comment:
Can we be more specific about what exceptions we should handle here?

Reply:
I am following the pattern in the one at `_process_executor_events` method. We should improve this but I'm not sure the exceptions and not taking chances